### PR TITLE
Remove design docs page

### PR DESCRIPTION
**Pull request by itsjustkevin**
Fixes: https://github.com/flutter/website/issues/8749

## Description

Removes the design docs page on Flutter.dev in favor of a new process. 

## What problem are we solving?
 
Flutter community processes contain multiple entry points, creating a fragmented contributor experience.  Contributors must be aware of various processes to contribute to Flutter.  In an effort to unify the contributor experience, reduce documentation and streamline processes, I am proposing that we utilize github issues to share design documents, aligning with other Flutter processes.

CC: @Hixie @parlough @sfshaza2 

## Remaining work

- [ ] Verify that no other removals are necessary.
- [ ] Update wiki to clarify new process.
- [ ] Announce the new process to contributors.
- [ ] Add redirect to link back to the github issues holding the design documents.

### CODE REVIEW

**Comment by itsjustkevin** [comment]
Looks like I am missing something in the configs for the actions workflow.  Will investigate.

File: firebase.json [hunk 1]
```diff
@@ -178,333 +178,8 @@
       { "source": "/research-signup", "destination": "https://docs.google.com/forms/d/e/1FAIpQLSe0i4De809KXVCdljGKrjMj3lxhuzbuFKCtY5PEQPCYtGxFMg/viewform?usp=sf_link", "type": 301 },
       { "source": "/resources/design-principles", "destination": "https://github.com/flutter/flutter/wiki/Style-guide-for-Flutter-repo", "type": 301 },
       { "source": "/style-guide*", "destination": "https://github.com/flutter/flutter/wiki/Style-guide-for-Flutter-repo", "type": 301 },
-      { "source": "/youtube", "destination": "https://youtube.com/@flutterdev", "type": 301 },
+      { "source": "/youtube", "destination": "https://youtube.com/@flutterdev", "type": 301 }
 
-      { "source": "/go/2D-Foundation", "destination": "https://docs.google.com/document/d/1C2hAq-gMAIx4Cbym7EMhVAcPZ2hmNxlgijpYmgTpmUU/edit?usp=sharing", "type": 301 },
```
Comment (parlough) [inline firebase.json:183]: I know this is a draft, but as the PR stands, I have some concerns about removing the existing go links. With them removed, links to design docs around the internet, Discord, etc will be broken. I believe a few don't link to design docs as well. Perhaps we can deprecate the process, but keep the existing links around for a certain mount of time or indefinitely?

File: firebase.json
Comment (parlough) [inline firebase.json:1]: Once ready, can you add a redirect from the old design doc page (`/resources/design-docs`) to the location describing the process? Or perhaps the new location for design docs on GitHub. Whichever you think is better.

**Review by parlough — COMMENTED** [review]
Thanks for working on this project/streamlining and removing the old page and JS code as part of this work!

As for the build breaking, it should be because the page is still referenced in the sidenav: https://github.com/flutter/website/blob/main/src/_data/sidenav.yml#L437-L438

File: firebase.json (hunk 1 above)
Comment (itsjustkevin) [reply firebase.json:183]: I am 1000% fine with this, this commit was me exploring what could be causing the breakage in the action.  Will revert!

Comment (parlough) [reply firebase.json:183]: Sounds good, thanks!

File: firebase.json
Comment (itsjustkevin) [reply firebase.json:1]: This is a good idea, I can add a redirect to the firebase rules to point back to a list of issues in github, adding that to the todos section.  Thanks @parlough!

**Comment by sfshaza2** [comment]
Wow, @parlough, even tells you the exact line to delete! Sweeeeet.

**Comment by Hixie** [comment]
Can we add a redirect for the old page that goes to https://github.com/flutter/flutter/issues?q=is%3Aopen+is%3Aissue+label%3A%22design+doc%22 ?

**Comment by parlough** [comment]
I think we could just permanently add a redirect from the old page in the `firebase.json` to the design doc process. There just make sure to clearly link to the list of current design documents.

**Comment by itsjustkevin** [comment]
> I think we could just permanently add a redirect from the old page in the `firebase.json` to the design doc process wiki page. There just make sure to clearly link to the list of current design documents.
> 
> Then we can use that redirect as a stable link destination in-case the process ever moves locations.

Updated again, how does this look?

**Review by parlough — APPROVED** [review]
This looks good to me, thanks!

As long as this text from the [wiki page](https://github.com/flutter/flutter/wiki/Design-Documents):

> These links are automatically indexed on our web site: https://docs.flutter.dev/resources/design-docs

is removed before landing this.

**Comment by parlough** [comment]
Also, is there a plan to push current design doc authors to add to the issues?

**Comment by itsjustkevin** [comment]
Wiki page is updated. 

> Also, is there a plan to push pre-existing (and still ongoing) design doc authors to add to the issues?

@Hixie posted an announcement in the [Discord announce channel](https://discord.com/channels/608014603317936148/608116355836805126) asking authors to consider filling out an issue.

**Pull request merged**